src/main/java/jcifs/smb1/UniAddress.java

```diff
@@ -0,0 +1,470 @@
+                        "dns:/_ldap._tcp.dc._msdcs." + name,
+                        new String[] { "SRV" }
+                    );
+                    return name;
+                } catch (NameNotFoundException nnfe) {
+                    uhe = new UnknownHostException(nnfe.getMessage());
+                }
+                int dot = name.indexOf('.');
+                if (dot == -1)
+                    break;
+                name = name.substring(dot + 1);
+            }
+        } catch (NamingException ne) {
+            if (log.level > 1)
+                ne.printStackTrace(log);
+        }
+
+        throw uhe != null ? uhe : new UnknownHostException("invalid name");
+    }
+     */
+
+
+    Object addr;
+    String calledName;
+
+    /**
+     * Create a <tt>UniAddress</tt> by wrapping an <tt>InetAddress</tt> or
+     * <tt>NbtAddress</tt>.
+     */
+
+    public UniAddress( Object addr ) {
+        if( addr == null ) {
+            throw new IllegalArgumentException();
+        }
+        this.addr = addr;
+    }
+
+    /**
+     * Return the IP address of this address as a 32 bit integer.
+     */
+
+    public int hashCode() {
+        return addr.hashCode();
+    }
+
+    /**
+     * Compare two addresses for equality. Two <tt>UniAddress</tt>s are equal
+     * if they are both <tt>UniAddress</tt>' and refer to the same IP address.
+     */
+    public boolean equals( Object obj ) {
+        return obj instanceof UniAddress && addr.equals(((UniAddress)obj).addr);
+    }
+/*
+    public boolean equals( Object obj ) {
+        return obj instanceof UniAddress && addr.hashCode() == obj.hashCode();
+    }
+*/
+
+    /**
+     * Guess first called name to try for session establishment. This
+     * method is used exclusively by the <tt>jcifs.smb1.smb1</tt> package.
+     */
+
+    public String firstCalledName() {
+        if( addr instanceof NbtAddress ) {
+            return ((NbtAddress)addr).firstCalledName();
+        } else {
+            calledName = ((InetAddress)addr).getHostName();
+            if( isDotQuadIP( calledName )) {
+                calledName = NbtAddress.SMBSERVER_NAME;
+            } else {
+                int i = calledName.indexOf( '.' );
+                if( i > 1 && i < 15 ) {
+                    calledName = calledName.substring( 0, i ).toUpperCase();
+                } else if( calledName.length() > 15 ) {
+                    calledName = NbtAddress.SMBSERVER_NAME;
+                } else {
+                    calledName = calledName.toUpperCase();
+                }
+            }
+        }
+
+        return calledName;
+    }
+
+    /**
+     * Guess next called name to try for session establishment. This
+     * method is used exclusively by the <tt>jcifs.smb1.smb1</tt> package.
+     */
+
+    public String nextCalledName() {
+        if( addr instanceof NbtAddress ) {
+            return ((NbtAddress)addr).nextCalledName();
+        } else if( calledName != NbtAddress.SMBSERVER_NAME ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.